lib/mail/smtp_envelope.rb

```diff
@@ -43,14 +43,18 @@ def message=(message)
 
     private
       def validate_addr(addr_name, addr)
-        if addr.bytesize > MAX_ADDRESS_BYTESIZE
-          raise ArgumentError, "SMTP #{addr_name} address may not exceed #{MAX_ADDRESS_BYTESIZE} bytes: #{addr.inspect}"
-        end
-
         if /[\r\n]/ =~ addr
           raise ArgumentError, "SMTP #{addr_name} address may not contain CR or LF line breaks: #{addr.inspect}"
         end
 
+        if !addr.ascii_only? && Encodings.idna_supported?
+          addr = Address.new(addr).address_idna
+        end
```

Comment:
Again, could the `Address` class handle this without having to spread the idna checks in the code?